torch/csrc/jit/script/compiler.cpp

```diff
@@ -1748,37 +1748,34 @@ struct to_ir {
 
     int64_t index;
     if (idx_val->node()->kind() == prim::Constant) {
-      index = idx_val->node()->i(attr::value);
+      index = *constant_as<int64_t>(idx_val);
```

Comment:
This is going to silently fail if idx_value is not an integer constant.
```
if (auto value = constant_as<int64_t>(idx_val)) {
}
```

Reply:
Isn't the check that the node is a constant sufficient ?